### PR TITLE
feat: user login and sessions

### LINKED ISSUE #65: user session and login
## acceptance criteria
* As a user I can log in with an email and password
* I can also log out
* I can see my name in the dropdown menu in the top right
* My session is [secure](https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html). Most notably:
  * My password is salted and hashed
  * The session expires
  * The session renews automatically for longer run connections
  * The session is not remembered once I close

## notes
* This story won't have a way to create a user, so we will have to insert one into the database manually.
* Logging at this point won't change access

### PR DESCRIPTION
**Pull request by arrocke**
closes #65 

Emails won't send in the preview environment until our SES gets approved for production use. There are instructions in the docs for how to test locally

Changes
* Set up email authentication with NextAuth
* Show current user in profile dropdown
* Removed proxy in favor of requests directly to the api server with CORS support

### CODE REVIEW

File: docs/auth.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+We are using [NextAuth](https://next-auth.js.org/) to manage authentication.
+Authentication is handled on the api server since there are plans to have multiple apps with a common login.
+
+Presently only the [email provider](https://next-auth.js.org/providers/email) is supported. This provider works by sending temporary magic links in an email. When the user clicks the link, they are authenticated.
+
+Only existing users are able to log in. New users must be invited by a system or language admin.
+
+## Login in Dev Environments
+
+In local environments, we use [Ethereal Email](https://ethereal.email/) to capture emails without sending them. You can use the credentials in the EMAIL_SERVER env var in [packages/api/.env](../packages/api/.env) to log in and view sent emails.
```
Comment (Pertempto) [inline docs/auth.md:12]: Do I need to use extra steps to setup in my dev environment? I tried clicking the "sign in with email" button, and got this:  
image

image

Comment (Pertempto) [reply docs/auth.md:12]: @arrocke

**Comment by arrocke** [comment]
I pushed up a commit to fix this. You will need to insert a user into the database manually before login will work

**Review by Pertempto — APPROVED** [review]
It works well! The styling differences between the frontend and the login page are a little jarring, but that isn't very important at this point.

**Comment by arrocke** [comment]
That's the default for NextAuth. I had everything working with consistent styles, but it didn't work on vercel because of cookie issues.

**Pull request merged**